### DISCUSSION TITLE
installation problems: wsl + python 3.8 / 3.9

### DISCUSSION THREAD

**Discussion by cunyizju** [idaholab/moose#19029] [Q&A Getting Started] [answered]
When I install moose in windows 10 - wsl (python 3.9), the following messages are met:

>>> conda install moose-tools
```
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: \
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versionsThe following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.31=0
  - feature:|@/linux-64::__glibc==2.31=0

Your installed version is: 2.31
```

conda install moose-libmesh (py3.9):
```
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: |
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package libgcc-ng conflicts for:
python=3.9 -> zlib[version='>=1.2.11,<1.3.0a0'] -> libgcc-ng[version='>=7.2.0']
moose-libmesh -> moose-libmesh-vtk==6.3.0=build_7 -> libgcc-ng[version='>=7.2.0|>=9.3.0|>=7.3.0']
python=3.9 -> libgcc-ng[version='>=7.3.0|>=7.5.0']The following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.31=0
  - feature:|@/linux-64::__glibc==2.31=0
  - python=3.9 -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']

Your installed version is: 2.31
```
Similar problems occur for the python 3.8 environment.
Could anyone please help me check how to fix the above installation problem?

Best wishes,
Bruce

**Comment by GiudGiud** [comment, marked as answer]
@milljm

**Reply by cunyizju** [reply]
Let me answer this question on my own.

```
conda create -n moose -q -y
conda config --add channels idaholabCopy
conda config --add channels conda-forge

conda activate moose
conda install moose-tools
conda install moose-libmesh
```
should work.

**Reply by milljm** [reply]
The idaholab channel is being deprecated... It would seem I forgot to modify our FAQ section reflecting this change. While 'idaholab' still holds the necessary things, it won't forever. Also looks like the 'right-click select all' routine includes 'Copy' by mistake. The proper channel is our new public URL (https://conda.software.inl.gov/public):

```bash
conda config --add channels https://conda.software.inl.gov/public
conda config --remove channels idaholab # or idaholabCopy in your case
```
These commands are safe to use, as both channels contain identical packages.

As for your original error, I suspect a new environment would work (you're heading in the right direction).

**Reply by cunyizju** [reply]
Thanks, Jason. 
I believe creating a clean environment is critical to avoid conflicts.
```
conda create -n moose -q -y
```

**Reply by milljm** [reply]
Can you post what you did in a new post (not a reply). I'd like to mark it as the answer, especially since you figured it out on your own. It will help others that come across this same error :)

Cheers!

**Reply by rlt97** [reply]
Hi all, 

Thanks for your insight  @milljm and @cunyizju !

I am currently running into a similar issue regarding the installation of the moose packages 

>  C:\Users\lethai>conda activate moose

> (moose) C:\Users\lethai>conda install moose-tools
> Collecting package metadata (current_repodata.json): done
> Solving environment: failed with initial frozen solve. Retrying with flexible solve.
> Collecting package metadata (repodata.json): done
> Solving environment: failed with initial frozen solve. Retrying with flexible solve.

> PackagesNotFoundError: The following packages are not available from current channels:

>  - moose-tools

> Current channels:

  > - https://conda.anaconda.org/idaholab/win-64
  > - https://conda.anaconda.org/idaholab/noarch
  >- https://conda.software.inl.gov/public/win-64
 > - https://conda.software.inl.gov/public/noarch
 > - https://conda.anaconda.org/conda-forge/win-64
 > - https://conda.anaconda.org/conda-forge/noarch
>  - https://repo.anaconda.com/pkgs/main/win-64
>  - https://repo.anaconda.com/pkgs/main/noarch
>  - https://repo.anaconda.com/pkgs/r/win-64
 > - https://repo.anaconda.com/pkgs/r/noarch
 > - https://repo.anaconda.com/pkgs/msys2/win-64
 > - https://repo.anaconda.com/pkgs/msys2/noarch

If you have any hints, it will be greatly appreciated.

Thanks in advance, 

Remi

**Reply by milljm** [reply]
Unfortunately we do not have a Microsoft Windows channel at this time... You'll have to use Windows Subsystem for Linux (WSL) if you wish to make use of the MOOSE framework. Native Windows support is something we talk about on occasion, so perhaps someday!

**Reply by rlt97** [reply]
Got it - thanks ! @milljm

**Comment by LiHao1995** [comment]
Hello everyone. When installing metawrap, even though I created the virtual environment metawrap, there were still conflicts.
As following: 
(metawrap) [localhost ~]$ conda install -c ursky metawrap-mg
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: \ 
Found conflicts! Looking for incompatible packages.                                                                                                                                                                                                                                    failed                                                                                                                                                                                                                                                                                     

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versionsThe following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.18=0
  - python=2.7 -> libgcc-ng[version='>=11.2.0'] -> __glibc[version='>=2.17']

Your installed version is: 2.18

However, I successfully reinstalled glibc 2.18 and upgraded libgcc-ng to the latest version, but the issue is still not resolved. Can anyone provide some better advice? Thank you very much.

**Reply by milljm** [reply]
I think your best bet would be to ask the maintainers of Metawrap at the feed stock repo: https://github.com/conda-forge/metawrap-feedstock. However, it doesn't seem to be very active.

Following all the links, I end up at https://github.com/jakirkham/metawrap for its homepage. And it looks like no one is home. I am not sure how much support you are going to find...

You may have better luck downloading the source from: https://github.com/jakirkham/metawrap and building it manually. Or attempting to modify the Feedstock recipe to grab the latest version from source and build the package yourself.

Sorry but I don't think we're going to be of much help beyond what I've suggested. Is this in some way related to MOOSE?

**Comment by LiHao1995** [comment]
Thanks for your valuable advice ！ I will try the next step. If it fails again, metarap will have to be replaced with other software, as it is not the only one available. [ 0_.._0 ]









At 2023-04-20 20:27:21, "Jason Miller" ***@***.***> wrote:

I think your best bet would be to ask the maintainers for Metawrap at the feed stock repo: https://github.com/conda-forge/metawrap-feedstock. However, it doesn't seem to be very active.

Following all the links, I end up at https://github.com/jakirkham/metawrap for its homepage. And it looks like no one is home. I am not sure how much support you are going to find...

You may have better luck downloading the source from: https://github.com/jakirkham/metawrap and building it manually. Or attempting to modify the Feedstock recipe to grab the latest version from source and build the package your self.

Sorry but I don't think we're going to be of much help beyond what I've suggested. Is this in some way related to MOOSE?
